src/lib.rs

```diff
@@ -223,6 +221,28 @@ impl LogicalClock {
         overflow
     }
 
+    /// Calculate the clock which follows this one.
+    pub fn next(&self) -> LogicalClock {
+        let mut new = *self;
+        new.increment();
+        new
+    }
+
+    /// Calculate the clock which preceeded this one.
+    pub fn prev(&self) -> LogicalClock {
```

Comment:
Could you add some testing around the next/prev mechanisms here showing they hold up across the overflow boundaries?